[MIG] l10n_ar_perceptions_basic

Tlki opened the pull request:
[ADD] unit test for whole l10n_ar_perceptions_basic
[DEL] **account_analytic_id** from p**erception.perception**
[ADD] option to archive perceptions
[ADD] used perceptions cannot be removed

Tlki committed:
[MIG] l10n_ar_perceptions_basic

gabrielo77 commented on l10n_ar_perceptions_basic/models/account_invoice.py:
```diff
@@ -137,7 +136,7 @@ def _onchange_partner_id(self):
             upd_lst = []
             for p in perception_ids:
```
This could be a list comprehension, is more readable.

gabrielo77 commented on l10n_ar_perceptions_basic/models/perception.py:
```diff
@@ -60,7 +61,47 @@ class PerceptionPerception(models.Model):
-        'account.analytic.account',
-        string='Analytic Account',
-    )
+    active = fields.Boolean('Active', default=True, store=True, readonly=False)
```
`store` and `readonly` parameters are redundant. Please remove them

gabrielo77 commented on l10n_ar_perceptions_basic/models/perception.py:
```diff
@@ -60,7 +61,47 @@ class PerceptionPerception(models.Model):
+                    if not res:
+                        continue
+                msg = 'Can not delete the record %s\
+                       because is used in the system.'
```
```
msg = 'Can not delete the record %s\
                       because is used in the system.'
```
msg = ("Can not delete the record %s"
            "because is used in the system")

Is better because you avoid the extra spaces at the beginning of **because is used...**

Also a better error message would be appreciated, i.e. which relational field is causing the problem and which record is related, etc. _in the system_ is not very descriptive nor helpful, if nothing at all.

gabrielo77 commented on l10n_ar_perceptions_basic/tests/test_account_invoice.py:
```diff
@@ -0,0 +1,77 @@
+        perception.date = False
+        account_invoice.date_invoice = '2019-06-27'
+        account_invoice.finalize_invoice_move_lines({})
+        assert perception.date == account_invoice.date_invoice
```
Please add assertion error messages which describe why the test failed in all the `assert` sentences.

gabrielo77 commented on l10n_ar_perceptions_basic/models/perception.py:
```diff
@@ -60,7 +61,47 @@ class PerceptionPerception(models.Model):
+                        query = """
+                            SELECT * FROM
+                        """
+                        query += field.relation_table
```
DO NOT ADD STRINGS. Python comes with a lot of builtin tools to perform string concatenation which are:

- more readable
- more efficient
- more pythonic
- easier to maintain in the long term

Tlki committed:
[IMP] code improved

[ADD] code comments to asserts error
[IMP] string concatenation
[DEL] store and readonly from active field
[ADD] list comprehension where possible
